pkg/runtime.v2/framework/core/framework.go

```diff
@@ -104,10 +104,10 @@ func (f *Framework) RunCustomValidationPlugins(oldObj, newObj *kubeflowv2.TrainJ
 	return aggregatedWarnings, aggregatedErrors
 }
 
-func (f *Framework) RunComponentBuilderPlugins(ctx context.Context, info *runtime.Info, trainJob *kubeflowv2.TrainJob) ([]client.Object, error) {
+func (f *Framework) RunComponentBuilderPlugins(ctx context.Context, runtimeJobTemplateSpec interface{}, info *runtime.Info, trainJob *kubeflowv2.TrainJob) ([]client.Object, error) {
 	var objs []client.Object
```

Comment:
```suggestion
func (f *Framework) RunComponentBuilderPlugins(ctx context.Context, obj client.Object, info *runtime.Info, trainJob *kubeflowv2.TrainJob) ([]client.Object, error) {
```

Should we limit the object only for kubernetes object? The any is not safe and could decrease RunComponentBuilderPlugins API values.

Reply:
In general I agree with you, however, what if runtimeJobTemplateSpec doesn't represent the Kubernetes object, but it represents part of the spec ? In that case, the Kubernetes object could be constructed with TrainJob, Info, and runtimeJobTemplateSpec at the Build() stage.

Reply:
As we discussed offline with @tenzen-y, we will use client.Object here to reduce number of errors in the runtime framework.